openmetadata-ui/src/main/resources/ui/src/pages/AddObservabilityPage/DestinationFormItem/DestinationSelectItem/DestinationSelectItem.tsx

```diff
@@ -0,0 +1,294 @@
+    DESTINATION_DROPDOWN_TABS.internal
+  );
+  const [destinationOptions, setDestinationOptions] = useState(
+    DESTINATION_SOURCE_ITEMS.internal
+  );
+
+  // Selected destinations list
+  const selectedDestinations = Form.useWatch<ModifiedDestination[]>(
+    'destinations',
+    form
+  );
+
+  // Selected source for the alert
+  const [selectedSource] =
+    Form.useWatch<CreateEventSubscription['resources']>(['resources'], form) ??
+    [];
+
+  const handleTabChange = useCallback(
+    (key) => {
+      setActiveTab(key);
+      setDestinationOptions(getFilteredDestinationOptions(key, selectedSource));
+    },
+    [selectedSource]
+  );
+
+  const destinationType = form.getFieldValue([
+    'destinations',
+    id,
+    'destinationType',
+  ]);
+  const subscriptionType = form.getFieldValue(['destinations', id, 'type']);
+
+  // Check if selected destination type is internal destination
+  const isInternalDestinationSelected = useMemo(
+    () => checkIfDestinationIsInternal(destinationType),
+    [destinationType]
+  );
+
+  const getTabItems = useCallback(
+    (children: ReactElement) =>
+      map(DESTINATION_DROPDOWN_TABS, (tabName) => ({
+        key: tabName,
+        label: (
+          <span data-testid={`tab-label-${tabName}`}>
+            {t(`label.${tabName}`)}
+          </span>
+        ),
+        children,
+      })),
+    []
+  );
+
+  const customDestinationDropdown = useCallback(
+    (menu: ReactElement, name: number) => {
+      return (
+        <Tabs
+          centered
+          activeKey={activeTab}
+          className="destination-select-dropdown"
+          data-testid={`destination-category-dropdown-${name}`}
+          defaultActiveKey={DESTINATION_DROPDOWN_TABS.internal}
+          items={getTabItems(menu)}
+          key={`destination-tabs-${name}`}
+          tabBarStyle={{
+            background: WHITE_COLOR,
+          }}
+          onTabClick={handleTabChange}
+        />
+      );
+    },
+    [handleTabChange, getTabItems, activeTab]
+  );
+
+  const getHiddenDestinationFields = (
+    isInternalDestination: boolean,
+    item: number,
+    destinationType: string
+  ) => (
+    <>
+      <Form.Item
+        hidden
+        initialValue={
+          isInternalDestination
+            ? destinationType
+            : SubscriptionCategory.External
+        }
+        key={`${destinationType}-category`}
+        name={[item, 'category']}
+      />
+      {!isInternalDestination && (
+        <Form.Item
+          hidden
+          initialValue={destinationType}
+          key={`${destinationType}-type`}
+          name={[item, 'type']}
+        />
+      )}
+    </>
+  );
+
+  const isEditMode = useMemo(() => !isEmpty(fqn), [fqn]);
+
+  // Using recommended method here https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
+  // To execute code after mounting instead of using useRef
+  const measuredRef = useCallback((node) => {
+    if (node !== null) {
```

Comment:
Not sure why we need this I checked the link from the comment and it's recommended to use this method for measuring DOM node 
But in the function I don't see we are measuring anything

Reply:
Sorry, the name is a bit misLeading.
So basically the code inside is supposed to be executed after the component is mounted. But in the mentioned link in the comment, its not recommended to use useRef for that logic, like we do elsewhere in the application (isMounted.current)..

Reply:
I would like to know more about this usecase. Not preferring to use above method either